packages/admin/src/pages/super-admin-dashboard/user/[id]/change-department.page.tsx

```diff
@@ -1,137 +1,92 @@
-  user: User;
-  departments: Department[];
-  csrfToken: string;
-};
+  return QuestionPageGetServerSideProps<
+    PageBodyResponse,
+    Awaited<ReturnType<typeof getChangeDepartmentPage>>,
+    typeof processPagePostResponse
+  >({
+    context,
+    fetchPageData,
+    processPagePostResponse,
+    jwt: getSessionIdFromCookies(context.req),
+    onErrorMessage: '',
+    onSuccessRedirectHref: `/super-admin-dashboard/user/${context.params?.id}`,
+  });
+}
 
-const UserPage = ({ departments, user, csrfToken }: UserPageProps) => {
+const UserPage = ({
+  csrfToken,
+  pageData,
+  formAction,
+}: InferProps<typeof getServerSideProps>) => {
+  const { user, departments } = pageData;
   const { publicRuntimeConfig } = getConfig();
   return (
     <>
       <Meta title="Manage User - Change Department" />
-      <div className="govuk-!-padding-top-2">
-        <div className="govuk-width-container">
-          <a
-            href={`${publicRuntimeConfig.SUB_PATH}/spadmin-dashboard`}
-            className="govuk-back-link"
-            data-cy="cy-back-button"
-          >
-            Back
-          </a>
-          <main className="govuk-main-wrapper govuk-main-wrapper--auto-spacing">
-            <div className="govuk-grid-row">
-              <div className="govuk-grid-column-two-thirds">
-                <span className="govuk-caption-l">{user.email}</span>
-                <h1 className="govuk-heading-l">
-                  Change the user&apos;s department
-                </h1>
-                <FlexibleQuestionPageLayout
-                  formAction=""
-                  csrfToken={csrfToken}
-                  fieldErrors={[]}
-                >
-                  <Radio
-                    fieldName="department"
-                    radioOptions={departments.map((department) => ({
-                      label: department.name,
-                      value: department.id,
-                    }))}
-                    fieldErrors={[]}
-                  />
-                  <div className="govuk-button-group">
-                    <a
-                      href=""
-                      role="button"
-                      draggable="false"
-                      className="govuk-button"
-                      data-module="govuk-button"
-                    >
-                      Change department
-                    </a>
-                    <a href="" className="govuk-link">
-                      Manage departments
-                    </a>
-                  </div>
-                </FlexibleQuestionPageLayout>
-              </div>
-            </div>
-          </main>
-        </div>
+
+      <CustomLink
+        isBackButton
+        href={`/super-admin-dashboard/user/${user.gap_user_id}`}
+      />
+
+      <div className="govuk-!-padding-top-7">
+        <FlexibleQuestionPageLayout
+          formAction={`${publicRuntimeConfig.SUB_PATH}${formAction}`}
+          csrfToken={csrfToken}
+          fieldErrors={[]}
```

Comment:
Forgot to implement field errors for when no department is selected - implementing now

Reply:
Turns out we don't want an error to be thrown in this scenario - but added functionality to handle fieldErrors if the BE returns any